main.cpp

```diff
@@ -1 +1,411 @@
+    std::string yString = "";
+    if (isComputer) {
+        
+        bool computerPlayOptimal;
+        if (computerPlayOptimal) {
+            std::array<std::string, 2> caseChosed = optimalAlgo();
+            xString = caseChosed[0];
+            yString = caseChosed[1];
+        } else {
+            //generate random coordinate
+            xString = std::to_string(rand() % GAME_ARR.size());
+            yString = std::to_string(rand() % GAME_ARR[0].size());
+        }
+    } else {
+        //ask user the coordinate of the case to check
+        std::string caseToAdd;
+        getline(std::cin, caseToAdd);
+
+        //get single coordinate x or y
+        xString = caseToAdd.substr(0, caseToAdd.find(','));
+        yString = caseToAdd.substr(caseToAdd.find(',') + 1);
+    }
+
+    //cast coordinate to int and get the asked case in the array
+    int &askedCase = GAME_ARR[std::stoi(xString, nullptr, 10)][std::stoi(yString, nullptr, 10)];
+    //check if the case hasn't been checked else return false
+    if (askedCase == 0) {
+        askedCase = user;
+        return true;
+    }
+    std::cout << "An error occurred, please retry" << std::endl;
+    return false;
+}
+
+/**
+ * This method display as user-friendly as possible the array of the game
+ */
+void displayArray() {
+    //iterate on each case of the array
+    for (int x = 0; x < GAME_ARR.size(); x++) {
+        for (int y = 0; y < GAME_ARR[0].size(); y++) {
+            switch (GAME_ARR[x][y]) {
+                case 2:
+                    std::cout << "  X  ";
+                    break;
+                case 1:
+                    std::cout << "  *  ";
+                    break;
+                case 0:
+                default:
+                    std::cout << " " << x << "," << y << " ";
+            }
+        }
+        std::cout << std::endl;
+    }
+}
+
+/**
+ * This method create a new game and describe all the process of a game until someone win
+ * @param isVsComputer if true user 2 is the computer
+ */
+void newGame(bool isVsComputer) {
+    bool currentUser = false; //The current user is 1 if false else 2 (or computer)
+    displayArray();
+    while (!(checkWinR(1) || checkWinR(2))) {
+        bool isPassed = false;
+        while (!isPassed) {
+            isPassed = addACase(currentUser ? 2 : 1, isVsComputer && currentUser);
+        }
+        displayArray();
+        currentUser = !currentUser;
+    }
+
+    if (checkWinR(1)) {
+        if (isVsComputer) {
+            std::cout << "Congratulation, you won !" << std::endl;
+        } else {
+            std::cout << "Congratulation, User 1 won !" << std::endl;
+        }
+    } else {
+        if (isVsComputer) {
+            std::cout << "Congratulation, you lost !" << std::endl;
+        } else {
+            std::cout << "Congratulation, User 2 won !" << std::endl;
+        }
+    }
+}
+
+
+int main() {
+    std::cout << "Hello, do you want to play against computer (y to proceed) ?" << std::endl;
```

Comment:
This feels like you don't have any other choice. It would be better to ask
`Play against [h]uman or [c]omputer?`